### PR TITLE
Fix source input edge case

### PR DESCRIPTION
**Pull request by gnawf**

### CODE REVIEW

File: test/src/test/resources/fixtures/new hydration/new batching/new-batching-absent-source-input.yml [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+          }
+        }
+      variables: { }
+    # language=JSON
+    response: |-
+      {
+        "data": {
+          "activity": [
+            {
+              "__typename__batch_hydration__content": "Activity"
```
Comment (andimarek) [inline test/src/test/resources/fixtures/new hydration/new batching/new-batching-absent-source-input.yml:113]: that is not a legal answer, is it? I mean there is no fragment/type selector involved and therefore every key must be present.

Comment (gnawf) [reply test/src/test/resources/fixtures/new hydration/new batching/new-batching-absent-source-input.yml:113]: Yeah I was creating tests with different theories with how the bug was occurring.

I just left it in anyway. I don't think there's much harm.

(I eventually just pulled in the data from the logs and got an exact replication).

**Review by sbarker2 — APPROVED** [review]

**Review by bbakerman — APPROVED** [review]
